docs/contributor/arch/028-network-policy-hardening.md

```diff
@@ -0,0 +1,242 @@
+
+1. **FluentBit Agent**
+   - **Network Policy Name:** `telemetry-fluent-bit`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 2020, 2021, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+2. **OTel Log Agent**
+   - **Network Policy Name:** `telemetry-log-agent`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8888, 13133, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+3. **OTel Metric Agent**
+   - **Network Policy Name:** `telemetry-metric-agent`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8888, 13133, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+4. **OTel Log Gateway**
+   - **Network Policy Name:** `telemetry-log-gateway`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8888, 13133, 4318, 4317, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+5. **OTel Metric Gateway**
+   - **Network Policy Name:** `telemetry-metric-gateway`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8888, 13133, 4318, 4317, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+6. **OTel Trace Gateway**
+   - **Network Policy Name:** `telemetry-trace-gateway`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8888, 13133, 4318, 4317, 15090(optional)
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+7. **Self Monitor**
+   - **Network Policy Name:** `telemetry-self-monitor`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 9090
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+8. **Telemetry Manager**
+   - **Network Policy Name:** `telemetry-manager-manager`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8080, 8081, 9443
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+## Decision
+
+The current network policies are too weak. They do not meet the requirements described in https://github.com/kyma-project/kyma/issues/18818. These policies allow all IP addresses for both incoming and outgoing traffic. They only limit ports for incoming traffic, which means they can be made stronger. However, we will still need to allow any IP address (0.0.0.0/0) in some cases for incoming and outgoing rules. External services like CLS and Dynatrace, as well as Kube API server, use different IP address ranges that we cannot predict. This makes it hard to restrict outgoing traffic by IP address. Instead, we will restrict outgoing traffic by port number.
+
+We also decided to use `networking.kyma-project.io/metrics-scraping: allowed` label selector not only for RMA, but also for metric agent, self-monitoring, and customer-managed Prometheus deployments. Gardener system pods cannot be labeled in the zero-trust mode, so these pods must either be excluded from scraping.
```

Comment:
The last sentence seems to be missing the "or" part :D 
Alternatively, drop "either".

```suggestion
We also decided to use the label selector `networking.kyma-project.io/metrics-scraping: allowed` not only for RMA, but also for metric agent, self-monitoring, and customer-managed Prometheus deployments. Gardener system Pods cannot be labeled in the zero-trust mode, so these Pods must either be excluded from scraping [oooooor...?]
```